tests/api/user/delete.test.coffee

```diff
@@ -0,0 +1,32 @@
+CONFIG = require 'config'
+__ = CONFIG.universalPath
+_ = __.require 'builders', 'utils'
+should = require 'should'
+{ getUserGetter, customAuthReq } = require '../utils/utils'
+{ getRefreshedUser } = require '../fixtures/users'
+randomString = __.require 'lib', './utils/random_string'
+
+describe 'user:delete', ->
+  it 'should delete the user', (done)->
+    userPromise = getUserGetter(randomString(6), false)()
+    userPromise
+    .then (user)->
+      customAuthReq userPromise, 'delete', '/api/user'
+      .then (res)->
+        res.ok.should.be.true()
+        getRefreshedUser userPromise
+        .then (deletedUser)->
+          deletedUser._id.should.equal user._id
+          previousRevInteger = parseInt(user._rev .split('-')[0])
+          parseInt(deletedUser._rev.split('-')[0]).should.equal(previousRevInteger + 1)
+          deletedUser.username.should.equal user.username
```

Comment:
keeping username slot occupied ? what about i delete my account and someoneelse want to user de username i used

Reply:
it's not possible at the moment, neither before or after this PR, but having a look at it and the implications could be a todo